contracts/eosio.system/src/producer_pay.cpp

```diff
@@ -77,16 +77,28 @@ namespace eosiosystem {
       const asset token_supply   = eosio::token::get_supply(token_account, core_symbol().code() );
       const auto ct = current_time_point();
       const auto usecs_since_last_fill = (ct - _gstate.last_pervote_bucket_fill).count();
-
+      eosio::print("usecs_since_last_fill: ", usecs_since_last_fill);
       if( usecs_since_last_fill > 0 && _gstate.last_pervote_bucket_fill > time_point() ) {
          auto current_fees =  eosio::token::get_balance(token_account, fees_account, core_symbol().code() );
          auto distribute_tokens = static_cast<int64_t>( (continuous_rate * double(token_supply.amount) * double(usecs_since_last_fill)) / double(useconds_per_year) );
          auto fees_to_use = std::min( distribute_tokens, current_fees.amount );
          auto issue_tokens = distribute_tokens - fees_to_use;
          // needs to be 2/5 Savings, 2/5 Voters, 1/5 producers
-         auto to_per_block_pay = distribute_tokens / 5;
+         // add logic to calculate rng amount from issue_tokens, then subtract to get per_block_pay
+         auto rng_amount = issue_tokens * _gstate5.rng_rate / RATE_DENOMINATOR;
```

Comment:
Should it be `distribute_tokens` instead of `issue_tokens`?

Reply:
yes it should be distribute_tokens